cmd/main.go

```diff
@@ -36,7 +44,31 @@ var (
 )
 
 var saramaLogger = log.New(io.Discard, "[Sarama] ", log.LstdFlags)
+
+func InitTracer(JaegerExporterTracing bool) *sdktrace.TracerProvider {
+	resources, _ := resource.New(context.Background(),
+		resource.WithFromEnv(), // pull attributes from OTEL_RESOURCE_ATTRIBUTES and OTEL_SERVICE_NAME environment variables
+		resource.WithProcess(), // This option configures a set of Detectors that discover process information
+	)
+	//Could not use OTEL_TRACES_EXPORTER https://github.com/open-telemetry/opentelemetry-go/issues/2310
+	var exporter sdktrace.SpanExporter
+	if JaegerExporterTracing {
+		exporter, _ = jaeger.New(jaeger.WithAgentEndpoint()) //from env variable https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/jaeger#environment-variables
+	} else {
+		os.Getenv("HOST_IP")
+		exporter, _ = otlptracegrpc.New(context.Background(), otlptracegrpc.WithInsecure())
+	}
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithResource(resources),
+		sdktrace.WithBatcher(exporter),
+	)
+	otel.SetTracerProvider(tp)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}))
+	return tp
```

Comment:
Do we actually need to return the TracerProvider here?

Reply:
for shutdown yes